packages/botonic-cli/src/botonic-playground-service.ts

```diff
@@ -0,0 +1,98 @@
+import axios from 'axios'
+import localtunnel from 'localtunnel'
+
+import { GlobalCredentialsHandler } from './analytics/credentials-handler'
+import { PlaygroundSessionInfo } from './interfaces'
+import { getCurrentDirectory } from './util/file-system'
+
+const PLAYGROUND_API_HOST = 'https://api.playground.botonic.io'
+
+export class PlaygroundService {
+  baseUrl: string = PLAYGROUND_API_HOST
+  playgroundSessionsApi = `${this.baseUrl}/playground-sessions`
+  globalCredentialsHandler = new GlobalCredentialsHandler()
+  playgroundSession: PlaygroundSessionInfo | null
+  port: number
+  onClose?: () => void
+  tunnel: any
+
+  constructor(port: number, onClose?: () => void) {
+    this.port = port
+    this.onClose = onClose
+    this.tunnel = null
+  }
+
+  async start() {
+    try {
+      this.tunnel = await localtunnel({ port: this.port })
+      this.playgroundSession = await this.newPlaygroundSession({
+        anonymous_id: this.globalCredentialsHandler.getAnonymousId(),
+        url: this.tunnel.url,
+        app_name: getCurrentDirectory(),
+      })
+      this.onClose && this.tunnel.on('close', this.onClose)
+    } catch (e) {
+      console.log('Error creating local tunnel: ', e)
+    }
+  }
+
+  async getPlaygroundSession(
+    id: string
+  ): Promise<PlaygroundSessionInfo | null> {
+    if (!id) return null
+    try {
+      const resp = await axios.get(
+        `${PLAYGROUND_API_HOST}/playground-sessions/${id}/`
+      )
+      return resp.data as PlaygroundSessionInfo
+    } catch (e) {
+      return null
+    }
+  }
+
+  async newPlaygroundSession({
+    url,
+    app_name,
+    anonymous_id,
+  }): Promise<PlaygroundSessionInfo | null> {
+    try {
+      const resp = await axios.post(
+        `${PLAYGROUND_API_HOST}/playground-sessions/`,
+        {
+          anonymous_id,
+          url,
+          app_name,
+        }
+      )
+      return resp.data
+    } catch (e) {
+      console.log('Error creating Playground Session: ')
+      return null
+    }
+  }
+
+  get code(): string | undefined {
+    return this.playgroundSession?.code
+  }
+
+  async stop(): Promise<void> {
+    try {
+      this.tunnel && (await this.tunnel.close())
+      this.playgroundSession?.id &&
+        (await axios.delete(
+          `${PLAYGROUND_API_HOST}/playground-sessions/${this.playgroundSession.id}`
+        ))
+    } catch (e) {
+      console.log("Couldn't stop Playground session properly: ", e)
+    }
+  }
+
+  getTunnelSubdomain(url: string): string | null {
+    try {
+      const r = /https:\/\/(.*)\.loca\.lt/
+      return r.exec(url)![1]
+    } catch (e) {
+      return null
+    }
+  }
+}
```

Comment:
What's the purpose of this method?

Reply:
Ah, true, I did a big refactor and forgot to remove this function, thanks!